golang1.17/lib/launcher.go

```diff
@@ -206,6 +206,9 @@ func buildArguments(ctx context.Context, f reflect.Value, in []byte) ([]reflect.
 		return nil, nil
 	}
 
+	// Only build the context if the customer provided enough parameters in their function to require one.
+	ctx = buildContext(ctx)
```

Comment:
Why not call this above, where the deadline is put into the context as well?

Reply:
Decided to refactor. See below.